src/dotnet/APIView/APIViewWeb/Languages/CSharpLanguageService.cs

```diff
@@ -71,11 +71,13 @@ public override Task<CodeFile> GetCodeFileAsync(string originalName, Stream stre
                         {
                             using var nuspecStream = entry.Open();
                             var document = XDocument.Load(nuspecStream);
-                            foreach (var dependency in document.Descendants().Where(d => d.Name.LocalName == "dependency").Distinct())
-                            {
-                                dependencies.Add(
-                                    new DependencyInfo(dependency.Attribute("id").Value, dependency.Attribute("version").Value));
-                            }
+                            var dependencyElements = document.Descendants().Where(e => e.Name.LocalName == "dependency");
```

Comment:
Can you turn dependencies into objects first and then do the filtering? There are too many `e.Attribute("id").Value` all over the place